docs/react/guides/advanced-ssr.md

```diff
@@ -34,19 +34,37 @@ The first step of any React Query setup is always to create a `queryClient` and
 import { useState } from 'react'
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 
+function makeQueryClient() {
+  return new QueryClient({
+    defaultOptions: {
+      queries: {
+        // With SSR, we usually want to set some default staleTime
+        // above 0 to avoid refetching immediately on the client
+        staleTime: 60 * 1000,
+      },
+    },
+  })
+}
+
+let browserQueryClient: QueryClient | undefined = undefined
+
+function getQueryClient() {
+  if (typeof window === 'undefined') {
+    // Server: always make a new query client
+    return makeQueryClient()
+  } else {
+    // Browser: make a new query client if we don't already have one
+    // This is very important so we don't re-make a new client if React
+    // supsends during the initial render
+    if (!browserQueryClient) browserQueryClient = makeQueryClient()
+    return browserQueryClient
+  }
+}
+
 export default function Providers({ children }) {
-  const [queryClient] = useState(
-    () =>
-      new QueryClient({
-        defaultOptions: {
-          queries: {
-            // With SSR, we usually want to set some default staleTime
-            // above 0 to avoid refetching immediately on the client
-            staleTime: 60 * 1000,
-          },
-        },
-      }),
-  )
```

Comment:
I'm not sure these parts of the docs needs to change? Note that most of the examples on this page until the streaming library part uses `useQuery` and prefetching, which I don't think should be susceptible to this edge case? But I haven't thought about this as long as you have, so maybe I'm missing something?

Not something I feel strongly about though and happy to keep it as is to keep it aligned too.

If we do keep it as is, I would still love a comment here along the lines of "if you are using `useSuspenseQuery`, initiating the `QueryClient` as state can have edge cases, see the docs on streaming further down".

Reply:
I think it's important to put this in the docs in both places. I'm betting that many folks (like myself) will copy-paste the first example that they see, and having the `useState` in there will lead to a broken example **by default** until they add a suspense boundary.